### PR TITLE
Copy participants and ponderations from another setting

### PR DESCRIPTION
**Pull request by antopalidi**
#136 
Screenshot 2023-05-02 at 13 24 12

### CODE REVIEW

**Review by microstudi — CHANGES_REQUESTED** [review]
Good job @antopalidi 
let's change however some of the names to make it more clear. Let's go around the concept "copy from settings", instead of selected_settings or iterate around consultations directly.
See the comments.

File: app/controllers/decidim/action_delegator/admin/settings_controller.rb [hunk 1]
```diff
@@ -96,6 +96,14 @@ def settings
         def collection
           @collection ||= ActionDelegator::OrganizationSettings.new(current_organization).query
         end
+
+        def settings_select_options
+          collection.map { |setting| [setting.consultation.id, setting.consultation.title[I18n.locale.to_s]] }.to_h
```
Comment (microstudi) [inline app/controllers/decidim/action_delegator/admin/settings_controller.rb:101]:
```suggestion
          collection.map { |setting| [setting.consultation.id, translated_attribute(setting.title)] }.to_h
```

File: spec/commands/decidim/action_delegator/admin/create_setting_spec.rb [hunk 2]
```diff
@@ -38,4 +39,26 @@
...
+  context "when copy setting" do
+    let!(:copy_from_setting) { create(:setting, :with_participants, :with_ponderations, consultation: other_consultation) }
+    let(:other_consultation) { create(:consultation) }
+    let(:form) do
+      double(
+        invalid?: invalid,
+        max_grants: max_grants,
+        authorization_method: authorization_method,
+        decidim_consultation_id: decidim_consultation_id,
+        source_consultation_id: copy_from_setting.id
```
Comment (microstudi) [inline spec/commands/decidim/action_delegator/admin/create_setting_spec.rb:52]:
```suggestion
        copy_from_setting_id: copy_from_setting.id
```
There is something wrong here as this has been renamed. Test should fail if we were testing the if the participants where copied.

Can you try to add a test case where to check that participants have been copied?

File: spec/commands/decidim/action_delegator/admin/update_setting_spec.rb [hunk 3]
```diff
@@ -39,4 +40,32 @@
...
+  context "when copy setting" do
+    let(:copy_from_setting) { create(:setting, :with_participants, :with_ponderations, consultation: other_consultation) }
+    let(:other_consultation) { create(:consultation) }
+    let(:form) do
+      double(
+        invalid?: invalid,
+        max_grants: max_grants,
+        authorization_method: authorization_method,
+        decidim_consultation_id: decidim_consultation_id,
+        source_consultation_id: copy_from_setting.id
```
Comment (microstudi) [inline spec/commands/decidim/action_delegator/admin/update_setting_spec.rb:53]: same as before

File: spec/system/decidim/action_delegator/admin/admin_manages_settings_spec.rb [hunk 4]
```diff
@@ -41,6 +41,59 @@
...
+      fill_in :setting_max_grants, with: 5
+      select translated_attribute(consultation.title), from: :setting_decidim_consultation_id
+      select translated_attribute(first_consultation.title), from: :setting_copy_from_setting_id
+
+      find("*[type=submit]").click
+
+      expect(page).to have_admin_callout("successfully")
+      expect(page).to have_content(translated_attribute(first_consultation.title))
+      expect(page).to have_css("tr[data-setting-id='#{setting.id}'] td:nth-of-type(5)", text: "3")
+      expect(page).to have_css("tr[data-setting-id='#{setting.id}'] td:nth-of-type(6)", text: "3")
```
Comment (microstudi) [inline spec/system/decidim/action_delegator/admin/admin_manages_settings_spec.rb:67]: I don't understand this test, shouldn't be we checking the newly created settings if they have the same participants & ponderations as the old (but with new ids)

File: spec/commands/decidim/action_delegator/admin/create_setting_spec.rb (hunk 2 above)
Comment (antopalidi) [reply spec/commands/decidim/action_delegator/admin/create_setting_spec.rb]: Yes, you are right. I changed it and added new cases.

**Pull request merged**